server-nodejs/routes/permissions.js

```diff
@@ -3,32 +3,46 @@
  */
 
 module.exports = function (app) {
+    var tools = require('./perms-tools.js')(app);
+    var conf = app.locals.conf;
+
     app.use('/api/:route', function (req, res, next) {
-        switch (req.params.route) {
-        case 'lock':
-        case 'unlock':
-        case 'publish':
-        case 'upload':
-        case 'version':
-        case 'file':
-        case 'comment':
-            // User class must be at least 'user'
-            if (['user', 'editor', 'admin'].indexOf(req.user.class) === -1) {
+        var copyReq = Array.from(Array.isArray(req.body) ? req.body : [req.body]);
+
+        var isFromClass = tools.isFromClass(req);
+        if(['read', 'update'].indexOf(req.params.route) === -1) {
+            if(!isFromClass) {
                 return httpStatus(res, 403, 'Access ' + req.params.route);
             }
-            break;
-        case 'create':
-        case 'update':
-        case 'upsert':
-        case 'delete':
-            // User class must be at least 'editor'
-            if (['editor', 'admin'].indexOf(req.user.class) === -1) {
-                return httpStatus(res, 403, 'Edition');
-            }
-            break;
+            next();
+            return;
         }
-        next();
-    });
+
+        var ids = copyReq;
+        if(['update'].indexOf(req.params.route) > -1) {
+            ids = tools.extractIds(copyReq);
+        }
+        tools.filterByAuthor(ids, req.user.username, function(response) {
+            if(conf.authorMode) {
+                if(req.user.class === 'admin') {
```

Comment:
If understand well, here the goal is to override the filterByAuthor process if the user has his class which allows him/her to update/read everything. Thus, could we call here th method isFromClass for this purpose? This way, we would not need to hardcode the admin class name and it would work for any class we add (like editor)?